deps/parser/parser.jl

```diff
@@ -0,0 +1,203 @@
+    msg::String
+    UnparsablePolymakeFunction(function_name) = new("Cannot parse function: $function_name")
+end
+
+typestring = ""
+undotted_typestring = strip(typestring,'.')
+
+type_translate_list = Dict(
+    "BigObject"          => typestring * "pm_perl_Object",
+    "OptionSet"          => typestring * "pm_perl_OptionSet",
+    "Matrix"             => typestring * "pm_Matrix{T} where T",
+    "Matrix<Rational>"   => typestring * "pm_Matrix{" * typestring * "pm_Rational}",
+    "Matrix<Integer>"    => typestring * "pm_Matrix{" * typestring * "pm_Integer}",
+    "Vector"             => typestring * "pm_Vector{T} where T",
+    "Vector<Rational>"   => typestring * "pm_Vector{" * typestring * "pm_Rational}",
+    "Vector<Integer>"    => typestring * "pm_Vector{" * typestring * "pm_Integer}",
+    "Integer"            => typestring * "pm_Integer",
+    "Rational"           => typestring * "pm_Rational",
+    "Set"                => typestring * "pm_Set",
+    "Anything"           => "Any"
+)
+
+## Dummy, will soon be generated
+filenames_list = Array{String,1}(
+    [ abspath(joinpath("tmp","polytope.json")) ]
+)
+
+function julia_function_string(julia_name::String, polymake_name::String, app_name::String, julia_args::String, call_args::String, parameter_string::String )
+    return """
+function $julia_name( $julia_args )
+    $(typestring)application( "$app_name" )
+    return $(typestring)call_function( "$polymake_name$parameter_string", Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function julia_method_string(julia_name::String, polymake_name::String, app_name::String, julia_args::String, call_args::String, parameter_string::String )
+    return """
+function $julia_name( $julia_args )
+    $(typestring)application( "$app_name" )
+    return $(typestring)call_method( "$polymake_name$parameter_string", dispatch_obj, Array{Any,1}([ $call_args ]) )
+end
+
+"""
+end
+
+function create_argument_string_from_type( type_string::String, number::Int64 )
+    ## Special case for BigObject (we do not care 'bout the type here)
+    if startswith(type_string,"BigObject")
+        return "arg" * string(number) * "::" * type_translate_list["BigObject"]
+    end
+    ## We only translate the known cases
+    if !haskey(type_translate_list,type_string)
+        throw(UnknownPolymakeType(type_string))
+    end
+    return "arg" * string(number) * "::" * type_translate_list[type_string]
+end
+
+function parse_definition(method_dict::Dict, app_name::String)
+    name = method_dict["name"]
+    arguments = method_dict["args"]
+    mandatory = method_dict["mandatory"]
+    type_params = method_dict["type_params"]
+    is_method = haskey(method_dict,"method_of")
+    ## for now, we use the same name for Julia and Polymake
+    julia_name = name
+    polymake_name = name
+
+    ## Check for option set
+    has_option_set = last(arguments) == "OptionSet"
+
+    ## Make type parameters
+    param_list_header = Array{String,1}()
```

Comment:
this could be made into a separate function:
```julia
function params(n::Int)
    if n == 0
        return Array{String,1}(), ""
    end

    param_list_header = Array{String,1}()
    param_list = Array{String,1}()

    for i in 1:n
        push!(param_list_header,"param$i::String")
        push!(param_list,"\$param$i")
    end

    param_string = "<" * join(param_list,",") * ">"

    return param_list_header, param_string
end
```